lupa-core/src/main/kotlin/org/jetbrains/research/lupa/kotlinAnalysis/util/DatabaseConnection.kt

```diff
@@ -0,0 +1,64 @@
+package org.jetbrains.research.lupa.kotlinAnalysis.util
+
+import com.intellij.openapi.diagnostic.Logger
+import org.postgresql.util.PSQLException
+import java.io.FileInputStream
+import java.sql.Connection
+import java.sql.DriverManager
+import java.time.LocalDate
+import java.util.*
+
+class DatabaseConnection {
```

Comment:
I think you should make more extendable architecture:

```
abstract class DatabaseConnectionBase {
    protected var conn: Connection? = null
    protected val logger: Logger = Logger.getInstance(DatabaseConnectionBase::class.java)
}

open class PostgresDatabaseConnection(configEnv: String) : DatabaseConnectionBase() {
    init {
        val props = Properties()
        val propertiesFilePath = System.getenv(configEnv)

        propertiesFilePath?.let {
            props.load(FileInputStream(propertiesFilePath))

            Class.forName("org.postgresql.Driver")
            val url = "jdbc:postgresql://${props.getProperty("host")}/${props.getProperty("database")}"
            try {
                conn = DriverManager.getConnection(url, props.getProperty("user"), props.getProperty("password"))
                logger.info("Connected to database")
            } catch (e: PSQLException) {
                logger.error("Couldn't connect to database")
                e.printStackTrace()
            }
        }
    }
}

class KotlinTeamDatabaseConnection: PostgresDatabaseConnection("KOTLIN_DB_CONFIG") {

    fun updateRepoDate(repo: GitRepository) {}

    fun updateRepoDate(username: String, repoName: String) {}
    
    enum class RepositoriesTable(val value: String) {
        TABLE_NAME("kotlin_repositories_updates"),
        USERNAME_COL("username"),
        REPO_NAME_COL("repo_name"),
        ANALYSIS_DATE_COL("last_analysis_date"),
        PULL_DATE_COL("last_pull_date"),
    }
}
```

In this case, you can reuse DatabaseConnectionBase and PostgresDatabaseConnection for ither possible configs

Also, all KotlinTeam specific classes you should move into a separated package (from this file and from AnalysisExecutor)

What do you think?